Automattic/jetpack
# jetpack-mu-wpcom: migrate coming-soon to package

## Proposed changes:

- Branched from #28479
- Migrates the coming-soon feature to the `jetpack-mu-wpcom` package, that was previously added to the plugin version in #27815
- Adds initialization for the coming-soon feature.

### Other information:


## Jetpack product discussion

Internal PT: p9dueE-6jY-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

The `jetpack-mu-wpcom` package or plugin is not used in production anywhere just yet, so merging this should be safe and follow-up PRs can be used to address any issues that popup when implementing on wpcom/WoA. The bulk of the code in this PR is migration from #27815

I did do some quick testing on simple wpcom:

* `bin/jetpack-downloader test jetpack-mu-wpcom-plugin update/migrate-coming-soon-to-package`
* Will need to uncomment the require to `mu-wpcom-plugin.php` in `0-wpcom-jetpack-loader.php`.
* If not merged/deployed yet, will want to patch with changes in https://github.com/Automattic/wp-calypso/pull/72901
* Can then add some error logging to `prod/full-site-editing-plugin.php` & `class-jetpack-mu-wpcom.php` vendor package file to verify which codepath is being loaded.
* Enable the Coming Soon feature and test it out. Check PHP error log, there should be no redeclaration or other errors.
